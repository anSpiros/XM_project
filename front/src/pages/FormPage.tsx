import React from "react";
import Form from "../components/Form";
import { addResults } from "../state/actions/results";
import { RESULTS_ROUTE } from "../routes/routes";
import { useHistory } from "react-router-dom";
import { useDispatch } from "react-redux";

function FormPage() {
  let history = useHistory();
  const dispatch = useDispatch();
  const onSubmit = (data: any) => {
    dispatch(addResults(data));
    history.push(RESULTS_ROUTE);
  };
  return (
    <div className="row">
      <div className="col-12">
        <Form onSubmit={onSubmit} />
      </div>
    </div>
  );
}

export default FormPage;
