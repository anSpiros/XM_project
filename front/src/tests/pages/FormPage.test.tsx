import React from "react";
import { render } from "@testing-library/react";
import FormPage from "../../pages/FormPage";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { createStore } from "redux";
import allReducers from "../../state/reducers";

const store = createStore(allReducers);
describe("Form page component", () => {
  test("should render correctly", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <FormPage />
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
