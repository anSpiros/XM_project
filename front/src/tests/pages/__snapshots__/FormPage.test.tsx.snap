// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form page component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <form
        class="form-wrapper"
      >
        <div
          class="form-group"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-sm-6"
            >
              <label
                for="startDate"
              >
                Start date
              </label>
              <div
                class="DayPickerInput"
              >
                <input
                  class="form-control"
                  id="startDate"
                  name="startDate"
                  placeholder="YYYY-M-D"
                  value=""
                />
              </div>
            </div>
            <div
              class="col-12 col-sm-6"
            >
              <label
                for="endDate"
              >
                End date
              </label>
              <div
                class="DayPickerInput"
              >
                <input
                  class="form-control"
                  disabled=""
                  id="endDate"
                  name="endDate"
                  placeholder="YYYY-M-D"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <div
            class="row"
          >
            <div
              class="col-12 col-sm-6"
            >
              <label
                for="symbol"
              >
                Symbol
              </label>
              <select
                class="custom-select"
                id="symbol"
                name="symbol"
              >
                <option
                  disabled=""
                  selected=""
                  value=""
                >
                  Select
                </option>
                <option
                  value="AAPL"
                >
                  Apple Inc.
                </option>
                <option
                  value="MSFT"
                >
                  AppleMicrosoft Corporation
                </option>
                <option
                  value="AMZN"
                >
                  Amazon.com
                </option>
                <option
                  value="GOOGL"
                >
                  Alphabet Inc.
                </option>
                <option
                  value="FB"
                >
                  Facebook, Inc.
                </option>
                <option
                  value="TCEHY"
                >
                  Tencent Holdings Ltd.
                </option>
                <option
                  value="TSM"
                >
                  Taiwan Semiconductor
                </option>
              </select>
            </div>
            <div
              class="col-12 col-sm-6"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                type="email"
              />
            </div>
          </div>
        </div>
        <div
          class="form-group"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-6"
            >
              <button
                class="btn btn-primary w-100"
                data-testid="resolved"
                id="submit-btn"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
